packages/adapter-chiselstrike/src/index.ts

```diff
@@ -0,0 +1,191 @@
+import type { Account } from "next-auth";
+import type { Adapter, AdapterSession, AdapterUser, VerificationToken } from "next-auth/adapters"
+
+export class ChiselStrikeAuthFetcher {
+    url: string; /// ChiselStrike backend.
+    secret: string;
+
+    /// Makes a fetcher pointing to the ChiselStrike backend at this URL.  Use the secret provided by the ChiselStrike platform.
+    constructor(url: string, secret: string) {
+        this.url = url
+        this.secret = secret
+    }
+
+    /// Returns a ChiselStrike server auth path with this suffix.
+    private auth(suffix: string): string {
+        return `${this.url}/__chiselstrike/auth/${suffix}`
+    }
+
+    public sessions(suffix?: string): string {
+        return this.auth('sessions') + (suffix ?? '');
+    }
+
+    public accounts(suffix?: string): string {
+        return this.auth('accounts') + (suffix ?? '');
+    }
+
+    public users(suffix?: string): string {
+        return this.auth('users') + (suffix ?? '');
+    }
+
+    public tokens(suffix?: string): string {
+        return this.auth('tokens') + (suffix ?? '');
+    }
+
+    /// Like regular fetch(), but adds a header with this.secret.
+    public async fetch(input: RequestInfo, init?: RequestInit | undefined): Promise<Response> {
+        init ??= {}
+        init.headers = new Headers(init.headers)
+        init.headers.set('ChiselAuth', this.secret)
+        return fetch(input, init)
+    }
+
+    /// Fetches url and returns the first element of the resulting array (or null if anything goes wrong).
+    public async filter(url: string): Promise<null | { [_: string]: string }> {
+        const res = await this.fetch(url)
+        if (!res.ok) { return null }
+        const jres = await res.json()
+        if (!Array.isArray(jres.results) || jres.results.length < 1) { return null }
+        return jres.results[0]
+    }
+
+    public async deleteEverything() {
+        await this.fetch(this.sessions('?all=true'), { method: 'DELETE' })
+        await this.fetch(this.accounts('?all=true'), { method: 'DELETE' })
+        await this.fetch(this.users('?all=true'), { method: 'DELETE' })
+        await this.fetch(this.tokens('?all=true'), { method: 'DELETE' })
+    }
+}
+
+export function ChiselStrikeAdapter(fetcher: ChiselStrikeAuthFetcher): Adapter {
+    return {
```

Comment:
I don't see a need for a class here. I recommend changing the argument here to be a single object, and set up any helper methods before the `return` statement instead.